archive/mozilla_mar.ksy

```diff
@@ -0,0 +1,116 @@
+meta:
+  id: mozilla_mar
+  title: Mozilla ARchive
+  file-extension: mar
+  license: CC0-1.0
+  endian: be
+doc: |
+  Mozilla ARchive file is Mozilla's own archive format to distribute software updates.
+  Test files can be found on Mozilla's FTP site, for example:
+
+  <http://ftp.mozilla.org/pub/firefox/nightly/partials/>
+doc-ref: https://wiki.mozilla.org/Software_Update:MAR
+seq:
+  - id: magic
+    contents: "MAR1"
+  - id: ofs_index
+    -orig-id: OffsetToIndex
+    type: u4
+  - id: file_size
+    -orig-id: FileSize
+    type: u8
+  - id: len_signatures
+    -orig-id: NumSignatures
+    type: u4
+  - id: signatures
+    type: signature
+    repeat: expr
+    repeat-expr: len_signatures
+  - id: len_additional_sections
+    -orig-id: NumAdditionalSections
+    type: u4
+  - id: additional_sections
+    type: additional_section
+    repeat: expr
+    repeat-expr: len_additional_sections
+instances:
+  index:
+    pos: ofs_index
+    type: mar_index
+types:
+  signature:
+    seq:
+      - id: algorithm
+        -orig-id: SignatureAlgorithmID
+        type: u4
+        enum: signature_algorithms
+      - id: len_signature
+        -orig-id: SignatureSize
+        type: u4
+      - id: signature
+        size: len_signature
+  additional_section:
+    seq:
+      - id: len_block
+        type: u4
+      - id: block_identifier
+        -orig-id: BlockIdentifier
+        type: u4
+        enum: block_identifiers
+      - id: bytes
+        size: len_block - len_block._sizeof - block_identifier._sizeof
+        type:
+          switch-on: block_identifier
+          cases:
+            block_identifiers::product_information: product_information_block
```

Comment:
This should not be called `bytes` since it isn't always a byte array (if `block_identifier == block_identifiers::product_information`, you then access the `product_information_block` structure under property `bytes` which doesn't make any sense). It's common to call fields like that a `body`.